graph/src/ext/futures.rs

```diff
@@ -42,29 +42,47 @@ impl<F: Future, C: Fn() -> F::Error> Future for Cancelable<F, C> {
     }
 }
 
-#[derive(Debug)]
+/// A `CancelGuard` or `SharedCancelGuard`.
+pub trait CancelGuardTrait {
+    /// Adds `canceler` to the set being guarded.
+    fn add_canceler(&mut self, canceler: oneshot::Sender<()>);
+}
```

Comment:
It looks like CancelGuard is missing `is_canceled` -- should that go in the trait?

Reply:
`CancelGuard` is never canceled, it only cancels on drop.

Reply:
Makes sense!